include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -321,12 +322,7 @@ _ForwardIterator
 __pattern_fill_n(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _ForwardIterator __first, _Size __count,
                  const _T& __value)
 {
-    // TODO: is this new implementation are correct?
-    // Previously we hadn't hetero impl for __pattern_fill_n
-
-    return __pattern_walk1_n(__tag, ::std::forward<_ExecutionPolicy>(__exec),
-                             __par_backend_hetero::make_iter_mode<__par_backend_hetero::access_mode::write>(__first),
-                             __count, fill_functor<_T>{__value});
+    return __pattern_fill(__tag, ::std::forward<_ExecutionPolicy>(__exec), __first, __first + __count, __value);
```

Comment:
`__internal::__pattern_fill` ?

Reply:
Why do you need to explicitly specify the namespace `__internal` ?
We already in this namespace. And we have a lot of calls like this (without specify namespace `__internal`) for other patterns (~32).

Reply:
I agree with explicit namespace intent but please don't change it in this PR because other places follow the same pattern

Reply:
OK